baxter_sim_kinematics/src/arm_kinematics.cpp

```diff
@@ -245,8 +246,8 @@ bool Kinematics::loadModel(const std::string xml) {
  */
 bool Kinematics::readJoints(urdf::Model &robot_model) {
   num_joints = 0;
-  boost::shared_ptr<const urdf::Link> link = robot_model.getLink(tip_name);
-  boost::shared_ptr<const urdf::Joint> joint;
+  std::shared_ptr<const urdf::Link> link = robot_model.getLink(tip_name);
+  std::shared_ptr<const urdf::Joint> joint;
```

Comment:
How about lines like this?
This can support both melodic and kinetic.

```
#if ROS_VERSION_MINIMUN(1, 14, 0)  ;; melodic
  std::shared_ptr<const urdf::Link> link = robot_model.getLink(tip_name);
  std::shared_ptr<const urdf::Joint> joint;
#else
  boost::shared_ptr<const urdf::Link> link = robot_model.getLink(tip_name);
  boost::shared_ptr<const urdf::Joint> joint;
#endif
```

Reply:
STFM
If you want, I'll post the patch this evening